src/main.ts

```diff
@@ -77,4 +77,107 @@ export class Minifier {
       }
     }
   }
+
+  private nextChar(str: string): string {
+    if (str === '$') {
+      return '_';
+    }
+
+    if (str === '_') {
+      return 'a';
+    }
+    return String.fromCharCode(str.charCodeAt(0) + 1);
+  }
+
+  private checkReserved(str: string): boolean {
+    var reserved = [
+      'break',
+      'case',
+      'class',
+      'catch',
+      'const',
+      'continue',
+      'debugger',
+      'delete',
+      'do',
+      'else',
+      'export',
+      'extends',
+      'finally',
+      'for',
+      'function',
+      'if',
+      'import',
+      'in',
+      'instanceof',
+      'let',
+      'new',
+      'return',
+      'super',
+      'switch',
+      'this',
+      'throw',
+      'try',
+      'typeof',
+      'var',
+      'void',
+      'while',
+      'with',
+      'yield',
+      'enum',
+      'await',
+      'int',
+      'byte',
+      'char',
+      'goto',
+      'long',
+      'final',
+      'float',
+      'short',
+      'double',
+      'native',
+      'throws',
+      'boolean',
+      'abstract',
+      'volatile',
+      'transient',
+      'synchronized'
+    ];
+
+    function inArray(elem, arr) { return (arr.indexOf(elem) > -1); }
```

Comment:
doesnt add much value since it is usable only within the function. Either inline, or extract to a top-level function (and eventually move to an `utils.js`).